internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -1070,6 +1074,18 @@ func (r *BMACReconciler) reconcileSpokeBMH(ctx context.Context, log logrus.Field
 		// If none platform do not attempt to create spoke BMH
 		return reconcileComplete{}
 	}
+	skipSpokeBMH, err := r.isBaremetalPlatformWithoutMAPI(ctx, cd)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to determine if cluster deployment %s/%s is baremetal platform without MAPI capability", cd.Namespace, cd.Name)
+		if propagateError {
+			return reconcileError{err: err}
+		}
+		return reconcileComplete{}
+	}
+	if skipSpokeBMH {
+		log.Infof("Skipping spoke BareMetalHost reconcile for agent %s/%s since it's baremetal platform without MAPI capability", agent.Name, agent.Namespace)
```

Comment:
nit: from a user perspective we are skipping spoke BMH creation

Reply:
True, thanks!! Modified to say creation